apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/ws/WebSocketNetworkTransport.kt

```diff
@@ -286,6 +295,18 @@ private constructor(
     messages.trySend(Dispose)
   }
 
+  /**
+   * Close the connection to the server (if it's open).
+   *
+   * This can be used to force a reconnection to the server, for instance when new auth tokens should be passed to the headers.
+   *
+   * The given [reason] will be propagated to [Builder.reopenWhen] to determine if the connection should be reopened. If not, it will be
+   * propagated to any Flows waiting for responses.
+   */
+  fun closeConnection(reason: Throwable) {
```

Comment:
Chose this name because `reconnect` is not exactly what this does - but no strong opinion on it!

Reply:
Yep, you're right 👍